### ISSUE TITLE
Inconsistent Notification for Deleted Messages

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21641]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any chat.
2. Write a message "Test1" and delete it. No message notification is shown.
3. Write a message "Test2" and click "Reply Thread".
4. Delete the first message of the reply thread without writing anything, and wait a few seconds for the message to be deleted.
5. Observe that the [Deleted message] notification pops up  for Test2, but not for Test1.
## Expected Result: 
The notification for deleted messages should be consistent across all scenarios. Either the [Deleted message] notification should appear for both Test1 and Test2, or it should not appear for either since only one message was deleted with different steps. Inconsistencies in the notification system can be confusing for users
## Actual Result: 
The [Deleted message] notification pops up for Test2, but not for Test1. This inconsistency in the notification for deleted messages can cause confusion for users.
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.32.5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/2e2b9125-ec85-4a4d-a28e-cefa2ab147fd

[screen-capture (30).webm](https://github.com/Expensify/App/assets/43996225/4323b359-7d23-496b-ac99-7125026f2d7a)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @tewodrosGirmaA
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687283415992419

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by joekaufmanexpensify** [comment]
I don't think this is a bug. It's expected behavior. Our threads design treats a thread like any other chat/chat room. In the first situation, you can delete the message because it's just a message. 

But in the second you've created a thread room. The thread is a child report linked to the parent, and it's a chat room, like any other DM/room. So even if you delete the first message in the room, the room still exists (and you can see it in LHN). This is consistent with the fact that deleting the first message in another room doesn't delete the existence of that room.  

This is why `[deleted message]` appears in this situation. Because it now has a chat room linked to it.

**Comment by joekaufmanexpensify** [comment]
Closing as this isn't a bug. But LMK if you have any other questions!

**Comment by tewodrosGirmaA** [comment]
Hi @joekaufmanexpensify I conducted a test scenario where I did not delete a message from a thread, but only deleted it after entering and exiting the reply thread. I noticed that deleting the message in Test 2 had a different result than deleting it in Test 1. Specifically, in Test 2, the message displayed as "[Deleted message]" after navigating to and from the reply thread, but when I deleted it, nothing was displayed. 

In my opinion, this creates confusion for the user. If we can make the results consistent for both situations, it would be more convenient for the user. If this is not a bug but rather an expected behavior, then we can leave this issue closed on GitHub. For further details, please refer to the videos.
https://github.com/Expensify/App/assets/118253010/27fc0324-9480-436e-8507-9f8995c1a0e5

**Comment by joekaufmanexpensify** [comment]
@tewodrosGirmaA Thanks for following up! Yep, this is still working as expected. 

In your most recent video, you just navigated away from the thread. If you check your LHN after creating the thread, and clicking back into the parent room, you'll see the thread room is still there (because it still exists and is linked to the parent comment). That's why we continue to show `[Deleted message]`.

**Comment by tewodrosGirmaA** [comment]
Thanks @joekaufmanexpensify  for explaining  I have a clear understanding  now
